witness/src/Dscp/Snowdrop/Storage/Rocks.hs

```diff
@@ -0,0 +1,75 @@
+-- | Rocksdb storage adapter for SD.
+
+module Dscp.Snowdrop.Storage.Rocks (rocksDbActions) where
+
+import Codec.Serialise (Serialise, deserialiseOrFail, serialise)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.Map as M
+import qualified Database.RocksDB as Rocks
+import Snowdrop.Block (TipValue (..))
+import Snowdrop.Core (IdSumPrefixed (..), Prefix (..), ValueOp (..), changeSetToList)
+import Snowdrop.Execution (DbModifyActions (..), SumChangeSet (..), sumChangeSetDBA)
+import UnliftIO (MonadUnliftIO)
+
+import Dscp.DB.Rocks.Real
+import Dscp.Snowdrop.Configuration
+import Dscp.Snowdrop.Serialise ()
+
+-- TODO exceptions
```

Comment:
This is my comment, but it should be taken care of anyway. Please either check that it's all good and remove it, or create a task to improve backend exceptions (or something). I'd prefer option (1).

Reply:
Not addressed.